lib/ansible/plugins/filter/ipaddr.py

```diff
@@ -641,6 +646,22 @@ def ipaddr(value, query='', version=False, alias='ipaddr'):
     if version and v.version != version:
         return False
 
+    redundant_msg = "Filter '%s' will be deprecated in 2.10, as it is redundant and uses incorrect terminology please use first_usable instead" % (query)
+    network_msg = "Filter '%s' will be deprecated in 2.10, as it does not properly handle /32 networks, please use network_id instead" % (query)
+
+    deprecate_map = {
+        'address/prefix': network_msg,
+        'host/prefix': network_msg,
+        'network': network_msg,
+        'gateway': redundant_msg,
+        'gw': redundant_msg,
+        'hostnet': redundant_msg,
+        'router': redundant_msg
+    }
+
+    if deprecate_map.get(query):
+        display.deprecated(deprecate_map[query])
```

Comment:
`display.deprecated` takes a `version` kwarg that defines the version that the functionality will be removed in.  Please use this as it will help us audit the code later to find items needing removed.